bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,222 @@
+    private @Nullable Data data = new Data();
+    private @Nullable Settings settings = new Settings();
+
+    public WlanThermoEsp32Handler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(WlanThermoExtendedConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            if (!config.getUsername().isEmpty() && !config.getPassword().isEmpty()) {
+                AuthenticationStore authStore = httpClient.getAuthenticationStore();
+                authStore.addAuthentication(new DigestAuthentication(config.getUri(), Authentication.ANY_REALM,
+                        config.getUsername(), config.getPassword()));
+            }
+            pollingScheduler = scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+        } catch (URISyntaxException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Failed to initialize WlanThermo Nano: " + e.getMessage());
+        }
+    }
+
+    private void checkConnection() {
+        try {
+            if (httpClient.GET(config.getUri()).getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+                ScheduledFuture<?> oldScheduler = pollingScheduler;
+                if (oldScheduler != null) {
+                    oldScheduler.cancel(false);
+                }
+                pollingScheduler = scheduler.scheduleWithFixedDelay(this::update, 0, config.getPollingInterval(),
+                        TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "WlanThermo not found under given address.");
+            }
+        } catch (URISyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not connect to WlanThermo at " + config.getIpAddress() + ": " + e.getMessage());
+            ScheduledFuture<?> oldScheduler = pollingScheduler;
+            if (oldScheduler != null) {
+                oldScheduler.cancel(false);
+            }
+            pollingScheduler = scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            State s = wlanThermoEsp32CommandHandler.getState(channelUID, data, settings);
+            if (s != null) {
+                updateState(channelUID, s);
+            }
+        } else {
+            if (wlanThermoEsp32CommandHandler.setState(channelUID, command, data)) {
+                logger.debug("Data updated, pushing changes");
+                push();
```

Comment:
Since `handleCommand` is called from UI threads you shouldn't make them execute potentially long running tasks like http requests. I suggest calling `push` asynchronously to prevent the UI from becoming unresponsive.